walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,82 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
+
+  // has one slot for every minute of a day
+  // The minute is true if all attendees are available in that minute, false
+  // otherwise
+  private boolean[] availability = new boolean[TimeRange.END_OF_DAY + 1];
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Arrays.fill(availability,true);
+
+    for (Event event : events) {
+      if (isEventRelevant(event, request.getAttendees())) {
+        recordBusyTime(event.getWhen());
+      }
+    }
+
+    Collection<TimeRange> availableTimes = new LinkedList<TimeRange>();
+
+    int startIndex = 0;
```

Comment:
The numeric zero here is a bit of a magic number, and depends on an implementation detail of the library. Since the library provides:
TimeRange.START_OF_DAY

It would make more sense to use there here.

Reply:
Yep good point, done